### PR TITLE
Unified output for Zabbix or other machine-parsing integrations

### PR DESCRIPTION
**Pull request by nugget**
I expanded the Printer class with helper methods to unify all the places in the code where items and values are printed.  This allowed the addition of several conveniences:

1. There's now a single place (defined via constants) to determine column widths for the label, value, and sparkline displays for coordinated output across the various classes.

2. Added support for `--no-labels` and `--no-scale` output options along the same lines as the existing `--no-graphs` option.  This makes for easier machine-parsing of the values without the user having to pipe the output through `sed`, `tr`, `cut` and friends to suss out the value for processing.

3. Added support for "reverse" semantic threshold arrays for the sparkline generation. This for items where the "good" end of the graph is 100%.  It enables a sparkline graph to be shown for current battery charge, for example, where it'll display green for 100% and red for 1% which is the opposite behavior as expected for temperature graphs.  This is controlled simply by supplying a threshold array where the values are in descending order instead of ascending.

4. Lastly (but the whole reason for my efforts) is the addition of a `scan zabbix` command which emits a JSON block of scanned keys suitable for use by [Zabbix](https://zabbix.com) to auto-discover temperature items for monitoring.  I added a template file and agent config file to an `integrations` directory which utilize this command.

I realize this is a pretty large PR just going by files touched, but the bulk of the work is cosmetic only.  I hope it's manageable enough to be integrated into the main project.  Let me know if you have any concerns or questions.  I'm happy to address them.

Runtime:
```
$ istats cpu temp --no-label --no-graph
41.81°C
$ istats cpu temp --no-scale --no-graph
CPU temp:               40.81
$ istats cpu temp --value-only
40.56
```

### CODE REVIEW

**Comment by Chris911** [comment]
Thanks for your contribution @nugget. I'll have to do a more thorough later but in the meantime can you think of a way to support a `--format` option is multiple options? For the scan method as well as other outputs I'd rather just have support for `--format=zabbix` that handles all the formatting logic. 

Also I'm sure how Zabbix work and what the exact use-case it but note that `scan` is meant to run once, then `istats enable all` to enable all sensors and the use `istats all` to display all stats without requiring another (slow) scan.

**Comment by nugget** [comment]
Yeah, when you look into the changes you'll see that `scan zabbix` actually has its own behavior that's not entirely related to how a bare `scan` works.  It's not just a re-formatting of the scan data.

I wavered on whether or not `scan zabbix` should leverage the enabled/disabled settings for the various keys, and in the end I went the other direction. Basically it just spits out a list of all the code-defined keys found in the `sensors_name` data that respond on that machine.  It's not walking the alphabet like a full scan.

I'd imagine the most common use case for Zabbix monitoring will be with more automated deployment of the Zabbix agent and iStats, meaning there won't necessarily be a person manually vetting the keys and making a deliberate selection of thermal keys for that machine.

The use case is automatically adding any relevant thermal sensors when a macOS machine is being monitored by a Zabbix monitoring server.  The `scan zabbix` is just providing a list of meaningful keys for that specific host which the server will then monitor via the Zabbix agent also installed on the host.

File: lib/iStats/smc.rb [hunk 1]
```diff
@@ -122,9 +125,29 @@ def scan_supported_keys
...
+
+        characters = [('a'..'z'), ('A'..'Z'),(0..9)].map { |i| i.to_a }.flatten
+        characters.each {|l1|
+          characters.each {|l2|
+            characters.each {|l3|
+              key = "T#{l1}#{l2}#{l3}"
+              if (name(key) != 'Unknown') && (name(key) != '')
+                t = is_key_supported(key);
+                if (t > 0) 
+                    items.push('{"{#KEY}":"' + key + '","{#NAME}":"' + name(key) + '" }')
```
Comment (Chris911) [inline lib/iStats/smc.rb]: Is that really the best way to build a JSON object in Ruby? Looks a bit hacky with string concatenation.

**Comment by Chris911** [comment]
I tested your branch locally and everything looks good. Only have 1 comment about how you build the JSON object for the zabbix scan. 

Please cleanup the commits a little bit (there shouldn't be more than 2-3 commits for this change) and I'll merge this.

**Comment by nugget** [comment]
OK, I've replaced the zabbix output with a proper JSON.generate call.  I don't know how to clean up the commits in the pull request, short of closing this one out, doing some sort of cherry pick on my end to a new branch, and then creating a new pull request.  Is that what you'd like for me to do?  I'd assumed you'd just do a squash merge on the PR to condense it down to a single commit on your end.

**Comment by Chris911** [comment]
Sure I'll squash on my end. FYI, you can modify the commits by going to your branch and run `git rebase -i master`. You can then use the squash or fixup instructions to merge arbitrary commits.

**Comment by nugget** [comment]
Thanks, that's good to know.

**Comment by Chris911** [comment]
@nugget v1.5.0 includes this change.

**Pull request merged**